### PR TITLE
Alpha beta rule change

### PR DESCRIPTION
**Pull request by rachtibat**
Hi Chris,

Sebastian and I, we found an error in the AlphaBeta rule.
We fixed it using the code from innvestigate that Sebastian wrote some time ago.

Best greetings from Sebastian and Reduan

### CODE REVIEW

File: zennit/rules.py [hunk 1]
```diff
@@ -62,16 +62,31 @@ class AlphaBeta(LinearHook):
     beta: float, optional
         Multiplier for the negative output term.
     '''
-    def __init__(self, alpha=2., beta=1.):
+    def __init__(self, alpha=2., beta=1.): 
+        assert alpha > 0 and beta > 0
+        assert (alpha - beta) == 1
```
Comment (chr5tphr) [inline zennit/rules.py:67]: I am not a fan of assertions outside of testing code.
If you must add this check, check it with an if-condition and raise a proper ValueError with a proper message

File: zennit/rules.py [hunk 2]
```diff
@@ -62,16 +62,31 @@ class AlphaBeta(LinearHook):
...
+                lambda param: param.clamp(min=0) 
+                ],
+
+            output_modifiers=[lambda output: output] * 4,
+
             gradient_mapper=(lambda out_grad, outputs: [out_grad / stabilize(output) for output in outputs]),
-            reducer=(lambda inputs, gradients: inputs[0] * (alpha * gradients[0] + beta * gradients[1]))
+      
+            reducer=(lambda inputs, gradients: alpha * (inputs[0] * gradients[0] + inputs[1] * gradients[1]) 
+                - beta * (inputs[2] * gradients[2] + inputs[3] * gradients[3]))
```
Comment (chr5tphr) [inline zennit/rules.py:89]: How is this different from using the output modifier instead?
The output-modifier uses the full gradient information, thus getting all these parts correctly.
Did you check its difference, and whether it produces the same output?

**Review by chr5tphr — CHANGES_REQUESTED** [review]

File: zennit/rules.py (hunk 2 above)
Comment (rachtibat) [reply zennit/rules.py:89]: The alpha beta rule is defined in http://iphome.hhi.de/samek/pdf/MonXAI19.pdf page 13

There you see, that we do not clamp the outputs but the inner part of the summation.
The matrix multiplication is defined as 
sum_i (input_i * weight_ij)
If you use output modifier you get
[sum_i (input_i * weight_ij)]⁺
but we want
sum_i [(input_i * weight_ij)]⁺

Hope it clarifies

File: zennit/rules.py (hunk 1 above)
Comment (rachtibat) [reply zennit/rules.py]: Alright, changed!

File: zennit/rules.py (hunk 2 above)
Comment (chr5tphr) [reply zennit/rules.py:91]: okay, agreed

**Pull request merged**